pkg/tfgen/generate_nodejs.go

```diff
@@ -507,7 +507,7 @@ func (g *nodeJSGenerator) emitResourceType(mod *module, res *resourceType) (stri
 			outcomment = "/*out*/ "
 		}
 
-		w.Writefmtln("    public %sreadonly %s: pulumi.Output<%s>;",
+		w.Writefmtln("    public %sreadonly %s!: pulumi.Output<%s>;",
```

Comment:
necessary since we don't actually set the values directly in our constructors (its done using reflection in registerResource).  this tells TS that that's ok and the values will be set through some mechanism.